api/comment_threads.rb

```diff
@@ -66,7 +66,29 @@
 
 put "#{APIPREFIX}/threads/:thread_id" do |thread_id|
   filter_blocked_content params["body"]
-  thread.update_attributes(params.slice(*%w[title body pinned closed commentable_id group_id thread_type]))
+  updated_content = params.slice(*%w[title body pinned closed commentable_id group_id thread_type close_reason_code])
+  # If a close reason code is provided, save it. If a thread is being reopened, clear the closed_by flag
+  if params[:user_id]
+    if updated_content.has_key? CLOSED
+      if value_to_boolean(updated_content[CLOSED])
+        updated_content["closed_by"] = user
+      else
+        updated_content["closed_by"] = nil
+        updated_content["close_reason_code"] = nil
+      end
+    end
+    if updated_content.has_key? BODY and updated_content[BODY] != thread.body
+      edit_reason_code = params.fetch("edit_reason_code", nil)
+      thread.edit_history.build(
+        original_body: thread.body,
+        author: user,
+        reason_code: edit_reason_code,
+        editor_username: user.username,
+      )
+      thread.save
```

Comment:
Nit: I don't think we need to call `save` here, since `update_attributes` (a couple of lines below) already calls `save`. Removing the `save` call would also prevent edit history from getting stored if the update of attributes fails.